opal/mca/btl/tcp/btl_tcp.c

```diff
@@ -531,3 +533,64 @@ void mca_btl_tcp_dump(struct mca_btl_base_module_t* base_btl,
     }
 #endif /* OPAL_ENABLE_DEBUG && WANT_PEER_DUMP */
 }
+
+
+/*
+ * A blocking recv on a non-blocking socket. Used to receive the small
```

Comment:
This comment is no longer true.  The socket is blocking (but with a receive timeout) during the initial handshake.   Does it make sense to continue using this wrapper in the initial handshake receive?  Possibly, as you'd need a bit of a loop to handle EINTR/EAGAIN in the case of a signal.  But then you need to update the comment with an explanation that usually (but not always) the socket is in non-blocking.